lib/src/dartdoc_options.dart

```diff
@@ -705,115 +657,133 @@ abstract class DartdocSyntheticOption<T> implements DartdocOption<T> {
 typedef OptionGenerator = Future<List<DartdocOption>> Function(
     PackageMetaProvider);
 
-/// A [DartdocOption] that only contains other [DartdocOption]s and is not an
-/// option itself.
-class DartdocOptionSet extends DartdocOption<void> {
-  DartdocOptionSet(String name, ResourceProvider resourceProvider)
-      : super(
-            name, null, null, OptionKind.other, false, null, resourceProvider);
+/// This is a [DartdocOptionSet] used as a root node.
+class DartdocOptionRoot extends DartdocOptionSet {
+  DartdocOptionRoot(String name, ResourceProvider resourceProvider)
+      : super(name, resourceProvider);
+
+  late final ArgParser __argParser =
```

Comment:
Is the double underscore a pattern from before the null safety migration, or new?

Reply:
It's from before.  I wrote this when I was still trying to write Python in Dart.  I can change.